service/create/service.go

```diff
@@ -644,8 +645,9 @@ func (s *Service) Boot() {
 					}
 					if err := kmsKey.Delete(); err != nil {
 						s.logger.Log("error", errgo.Details(err))
```

Comment:
At the moment we go ahead, same as we do for other errors on deletion.

* If the resource is not there, we should just go ahead
* If there was a temporary error (say, a network error), we could retry

I'd say this is a more general issue we should tackle in the scheme of making the operator more resilient. There was talk of having a focus on this after we get the basic functionality in place.